app/screens/global_threads/threads_list/thread/thread.tsx

```diff
@@ -0,0 +1,253 @@
+            marginHorizontal: 12,
+            marginVertical: 2,
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 25, 'Light'),
+        },
+        message: {
+            color: theme.centerChannelColor,
+            fontSize: 15,
+            lineHeight: 20,
+        },
+        unreadDot: {
+            width: 8,
+            height: 8,
+            borderRadius: 4,
+            backgroundColor: theme.sidebarTextActiveBorder,
+            alignSelf: 'center',
+            marginTop: 5,
+        },
+        mentionBadge: {
+            width: 18,
+            height: 18,
+            borderRadius: 9,
+            backgroundColor: theme.buttonBg,
+            alignSelf: 'center',
+        },
+        mentionBadgeText: {
+            ...typography('Body', 25, 'SemiBold'),
+            alignSelf: 'center',
+            color: theme.buttonColor,
+        },
+    };
+});
+
+const Thread = ({author, channel, post, teammateNameDisplay, testID, thread}: Props) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const handleMarkdownSoftBreak = useCallback(() => {
+        return '\n';
+    }, []);
+
+    const showThread = useCallback(preventDoubleTap(() => {
+        fetchAndSwitchToThread(serverUrl, thread.id);
+    }), [serverUrl, thread.id]);
+
+    const showThreadOptions = useCallback(() => {
+        const passProps = {thread};
+        const title = isTablet ? intl.formatMessage({id: 'thread.options.title', defaultMessage: 'THREAD ACTIONS'}) : '';
+
+        if (isTablet) {
+            showModal(Screens.THREAD_OPTIONS, title, passProps, bottomSheetModalOptions(theme, 'close-thread-options'));
+        } else {
+            showModalOverCurrentContext(Screens.THREAD_OPTIONS, passProps);
+        }
+    }, [isTablet, theme, thread]);
+
+    const threadStarterName = displayUsername(author, intl.locale, teammateNameDisplay);
+    const testIDPrefix = `${testID}.${thread.id}`;
+
+    const needBadge = thread.unreadMentions || thread.unreadReplies;
+    let badgeComponent;
+    if (needBadge) {
+        if (thread.unreadMentions) {
+            badgeComponent = (
+                <View
+                    style={styles.mentionBadge}
+                    testID={`${testIDPrefix}.unread_mentions`}
+                >
+                    <Text style={styles.mentionBadgeText}>{thread.unreadMentions > 99 ? '99+' : thread.unreadMentions}</Text>
+                </View>
+            );
+        } else if (thread.unreadReplies) {
+            badgeComponent = (
+                <View
+                    style={styles.unreadDot}
+                    testID={`${testIDPrefix}.unread_dot`}
+                />
+            );
+        }
+    }
+
+    let name;
+    let postBody;
+    if (post?.deleteAt && post.deleteAt > 0) {
```

Comment:
again, this should be `if (!post || post.deleteAt > 0)` as the post may not even be present in the database as it got removed after it was deleted.